proposals/DOM-Parts-Imperative.md

```diff
@@ -8,43 +8,67 @@ A DOM part is represented by the `Part` interface and its sub interfaces.
 
 ```webidl
 interface Part {
-    attribute any value;
-    void commit();
+  attribute any value;
+
+  readonly attribute PartRoot root;
+  readonly attribute FrozenArray<DOMString> metadata;
+  void commit();
+};
+
+dictionary PartInit {
+  FrozenArray<DOMString> metadata;
 };
 
 interface NodePart : Part {
-    constructor(Node node);
-    readonly attribute Node node;
+  constructor(PartRoot root, Node node, optional PartInit init = {});
+
+  readonly attribute Node node;
 };
 
 interface AttributePart : Part {
-    constructor(Element element, DOMString qualifiedName, DOMString? namespace);
-    readonly attribute DOMString prefix;
-    readonly attribute DOMString localName;
-    readonly attribute DOMString namespaceURI;
+  constructor(
+      PartRoot root,
+      Element element,
+      DOMString qualifiedName,
+      optional DOMString? namespace = null,
+      optional PartInit init = {});
+
+  readonly attribute DOMString prefix;
+  readonly attribute DOMString localName;
+  readonly attribute DOMString namespaceURI;
```

Comment:
```suggestion
  readonly attribute DOMString namespaceURI;
  readonly attribute DOMString rawName;
```

Consider adding a field representing the attribute name as it was written in the original source, before it was lower cased.